server/src/main/java/org/elasticsearch/rest/action/document/RestUpdateAction.java

```diff
@@ -83,6 +83,12 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         }
 
         updateRequest.retryOnConflict(request.paramAsInt("retry_on_conflict", updateRequest.retryOnConflict()));
+        if (request.hasParam("version")) {
+            updateRequest.version(RestActions.parseVersion(request));
```

Comment:
We should throw ActionRequestValidationException (instead of UnsupportedOperationException) to return 400. How about this?

```java
if (request.hasParam("version") || request.hasParam("version_type")) {
	final ActionRequestValidationException versioningError = new ActionRequestValidationException();
	versioningError.addValidationError("internal versioning can not be used for optimistic concurrency control. " +
		"Please use `if_seq_no` and `if_primary_term` instead");
	throw versioningError;
}
```

Reply:
I updated it and I suppose it is to be aligned with `DeleteRequest` ? Maybe we should move it to the `validate()` later and make the `version` part codes consistent with the `DeleteRequest` as you mentioned before ?